pysat/tests/classes/cls_instrument_iteration.py

```diff
@@ -310,229 +312,226 @@ def test_set_bounds_with_frequency_and_width(self):
                       == pds.date_range(start, stop, freq='10D').tolist())
         return
 
-    @pytest.mark.parametrize("values", [(dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 3),
-                                         2, 2),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 4),
-                                         2, 3),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 5),
-                                         3, 1),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 17),
-                                         5, 1)
-                                        ])
+    @pytest.mark.parametrize(
+        "starts,stops,step,width",
+        [(dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 3), 2, 2),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 4), 2, 3),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 5), 3, 1),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 17), 5, 1)])
```

Comment:
`starts` is the same in all calls, doesn't need to be a parameter.

Reply:
We have an open issue in #863 that could address this.  I'm thinking largely in terms of code readability for this release vs major rewrite here

Reply:
Added todo statements for both issues

Reply:
As a comment on related discussion, the stop dates are tied to the step and width values with the intent that  the full range of bounds behaviors are tested. I'm ok with a todo note for improvements to come in the future. It could be replaced with a more systematic identification of cases and applied more generally, as noted in #902.